### DISCUSSION TITLE
如何添加Aplayer？

### DISCUSSION THREAD

**Discussion by scutLMG** [ppoffice/hexo-theme-icarus#869] [Q&A]
如题，想要往icarus中添加Aplayer，照着网络上的教程来，并没有显示出来，对前端不是很熟悉，而且项目好像变成jsx了，不知道往哪里添加文件修改。
如我在layout/layout.jsx中修改，在最后添加如下代码

```
<link rel="stylesheet" href="APlayer.min.css">
<div id="aplayer"></div>
<script src="APlayer.min.js"></script>
</body>
</html>;
```

那么对样式的修改

```
const ap = new APlayer({
    container: document.getElementById('player'),
    fixed: true,
    audio: [{
        name: 'name',
        artist: 'artist',
        url: 'url.mp3',
        cover: 'cover.jpg',
    }]
});
```
应该在哪个文件内进行呢？

**Comment by ppoffice** [comment]
@scutLMG 可以也放在`layout.jsx`内，但是需要修改成React支持的语法。

配置代码可以放到[source/js/main.js](https://github.com/ppoffice/hexo-theme-icarus/blob/4.3.0/source/js/main.js)，同时建议Aplayer的库放到[layout/common/scripts.jsx](https://github.com/ppoffice/hexo-theme-icarus/blob/4.3.0/layout/common/scripts.jsx)中。

**Reply by scutLMG** [reply]
非常感谢！我已经发现了，但是react没学过，大致看了下就尝试改了是可以的。不过因为aplayer的代码注入影响了其他page的js代码，所以我已经放弃使用，用hexo-tag-aplayer插件了，谢谢解答！
